### DISCUSSION TITLE
How to export static /pages/page.js to be served as /page.html (not /page)

### DISCUSSION THREAD

**Discussion by osnwt** [vercel/next.js#11045] [Help]
I am trying to export static site to a set of `*.html` files. They can later be served by any HTTP server. The problem is that from a client view everything works, and requests to `/somepage` result in fetching `/somepage.html`. But if I refresh such page to get it from server directly, I have 404 error unless I add the html extension like `/somepage.html`.

In same cases it is possible to instruct a server to try known extensions like `html` or `htm` for requests without extension (MultiViews for Apache). Sometimes it is not possible. Is there any recommended way to refer to pages with full names (with `.html` extensions) when exporting a static site?

PS. Sufficient option could be to generate files without extension, that is, not `page.html`, but just `page`. How?

**Comment by Timer** [comment]
This configuration would depend 100% on how you are hosting it. What are you using to serve these files?

**Comment by osnwt** [comment]
I am newbie with web apps. So I try different options to know how.

Particularly, I have a PHP Symfony-based backend with some API, and I tried to copy nextjs-generated static frontend files to Symfony's `public` directory to be served from there. Symfony does not map requests to `/page` into `/page.html` and returns 404. So I have to rename generated `page.html` to just `page`. Then it works.

Of course, I would hardly use such a mix of technologies. But as I said, I want to know more inner details. So perhaps I can tell next (maybe using custom webpack config) to not use `html` extension for generated static page files.
